wacz/main.py

```diff
@@ -171,41 +181,61 @@ def create_wacz(res):
 
     passed_pages_dict = {}
 
-    # If the flag for passed pages has been passed
+    # Handle pages
     if res.pages != None:
-        print("Validating passed pages.jsonl file")
-        passed_content = []
-        with open(res.pages, "rb") as fh:
-            for line in fh:
-                if not line:
-                    continue
-
-                try:
-                    line = line.decode("utf-8")
-                    passed_content.append(line)
-                except:
-                    print("Page data not utf-8 encoded, skipping", line)
-
-        # Create a dict of the passed pages that will be used in the construction of the index
-        passed_pages_dict = construct_passed_pages_dict(passed_content)
+        if res.copy_pages:
+            print("Copying passed pages.jsonl file to WACZ")
+
+            if not validate_pages_jsonl_file(res.pages):
+                print("Unable to create WACZ without valid pages.jsonl file, quitting")
+                wacz.close()
+                return 1
+
+            with open(res.pages, "rb") as fh:
+                with wacz.open(pages_jsonl, "w") as pages_file:
+                    shutil.copyfileobj(fh, pages_file)
+
+        else:
+            print("Validating passed pages.jsonl file")
+            passed_content = []
+            with open(res.pages, "rb") as fh:
+                for line in fh:
+                    if not line:
+                        continue
+
+                    try:
+                        line = line.decode("utf-8")
+                        passed_content.append(line)
+                    except:
+                        print("Page data not utf-8 encoded, skipping", line)
+
+            # Create a dict of the passed pages that will be used in the construction of the index
+            passed_pages_dict = construct_passed_pages_dict(passed_content)
 
     if res.extra_pages:
-        print("Validating extra pages file")
-        extra_page_data = []
-        with open(res.extra_pages) as fh:
-            data = fh.read()
-            for page_str in data.strip().split("\n"):
-                page_json = validateJSON(page_str)
-
-                if not page_json:
-                    print("Warning: Ignoring invalid extra page\n %s" % page_str)
-                    continue
-
-                extra_page_data.append(page_str.encode("utf-8"))
-
-        extra_pages_file = zipfile.ZipInfo(EXTRA_PAGES_INDEX, now())
-        with wacz.open(extra_pages_file, "w") as efh:
-            efh.write(b"\n".join(extra_page_data))
+        if res.copy_pages:
+            print("Copying passed pages.jsonl file to WACZ")
+            if validate_pages_jsonl_file(res.extra_pages):
```

Comment:
Maybe can add a print log to notify that extra pages are skipped.

Reply:
Added